### PR TITLE
executor: fix a nil point when @@tidb_enable_collect_execution_info is off and cop cache is on

### LINKED ISSUE #35829: Unit test case 'executor TestIndexMergeRuntimeStats' fail on real TiKV
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

I'm [testing paging protocol](https://github.com/pingcap/tidb/issues/35633) by running tidb UT with a real TiKV using this branch https://github.com/pingcap/tidb/pull/35647

```
tiup playground nightly --mode tikv-slim
cd executor
go test -run TestIndexMergeRuntimeStats -with-tikv 127.0.0.1:2379
```

### 2. What did you expect to see? (Required)

No error

### 3. What did you see instead (Required)

```
--- FAIL: TestIndexMergeRuntimeStats (3.79s)
    testkit.go:174: 
        	Error Trace:	testkit.go:174
        	            				testkit.go:168
        	            				testkit.go:108
        	            				executor_test.go:3070
        	Error:      	Received unexpected error:
        	            	runtime error: invalid memory address or nil pointer dereference
        	            	github.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleTask.func1
        	            		/home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:664
        	            	runtime.gopanic
        	            		/home/genius/project/go/src/runtime/panic.go:838
        	            	runtime.panicmem
        	            		/home/genius/project/go/src/runtime/panic.go:220
        	            	runtime.sigpanic
        	            		/home/genius/project/go/src/runtime/signal_unix.go:818
        	            	github.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleCopResponse
        	            		/home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:961
        	            	github.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleTaskOnce
        	            		/home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:786
        	            	github.com/pingcap/tidb/store/copr.(*copIteratorWorker).handleTask
        	            		/home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:674
        	            	github.com/pingcap/tidb/store/copr.(*copIteratorWorker).run
        	            		/home/genius/project/src/github.com/pingcap/tidb/store/copr/coprocessor.go:416
        	            	runtime.goexit
        	            		/home/genius/project/go/src/runtime/asm_amd64.s:1571
        	Test:       	TestIndexMergeRuntimeStats
        	Messages:   	sql:select /*+ use_index_merge(t1, primary, t1a) */ * from t1 where id < 2 or a > 4 order by a, args:[]
FAIL
```

I've verified it's not caused by the changes from paging... instead, it's a bug on execution_info with cop cache.

### 4. What is your TiDB version? (Required)

master 

```
commit 47a4f3a017768dda0b9a826451486c5539bc8427 (HEAD -> master, origin/master, origin/HEAD)
Author: Weizhen Wang <wangweizhen@pingcap.com>
Date:   Wed Jun 29 20:16:40 2022 +0800

    *: enable unconvert (#35821)
    
    ref pingcap/tidb#35345
```

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #35829

Problem Summary:

### What is changed and how it works?

When `@@tidb_enable_collect_execution_info` is off,  the `resp.detail` is nil
And when cop cache code use it in `resp.detail.TimeDetail.ProcessTime`, tidb get a nil point panic.

### Check List

Tests 

- [ ] Unit test
- [X] Integration test
Run the test case with real tikv
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix a nil point panic when @@tidb_enable_collect_execution_info is off and cop cache feature is on. This combination is not the default setting, so most of the time a user would not meet it.
```

### CODE REVIEW

File: store/copr/coprocessor.go [hunk 1]
```diff
@@ -958,17 +958,19 @@ func (worker *copIteratorWorker) handleCopResponse(bo *Backoffer, rpcCtx *tikv.R
...
-			if worker.store.coprCache.CheckResponseAdmission(resp.pbResp.Data.Size(), resp.detail.TimeDetail.ProcessTime) {
-				data := make([]byte, len(resp.pbResp.Data))
-				copy(data, resp.pbResp.Data)
-
-				newCacheValue := coprCacheValue{
-					Data:              data,
-					TimeStamp:         worker.req.StartTs,
-					RegionID:          task.region.GetID(),
-					RegionDataVersion: resp.pbResp.CacheLastVersion,
+			if resp.detail != nil {
```
Comment (xhebox) [inline store/copr/coprocessor.go:961]: Maybe we can move `resp.detail != nil` into L960?

**Review by xhebox — APPROVED** [review]
LGTM

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**